iroh-unixfs/src/hamt.rs

```diff
@@ -74,54 +71,48 @@ impl Hamt {
         padding.len()
     }
 
-    pub fn children<'a, 'b: 'a, C: ContentLoader>(
+    #[allow(clippy::needless_lifetimes)]
+    pub fn children<'a, C: ContentLoader>(
         &'a self,
         ctx: LoaderContext,
-        loader: &'b Resolver<C>,
+        loader: C,
     ) -> impl Stream<Item = Result<Link>> + 'a {
         self.root.children(ctx, loader)
     }
 }
 
 impl InnerNode {
     pub async fn load_from_link<C: ContentLoader>(
-        ctx: crate::resolver::LoaderContext,
+        ctx: LoaderContext,
         link: &Link,
-        loader: &Resolver<C>,
+        loader: C,
     ) -> Result<Self> {
-        let path = Path::from_cid(link.cid);
-        let out = loader.resolve_with_ctx(ctx, path).await?;
-
-        match out.content {
-            OutContent::Unixfs(value) => match value {
-                UnixfsNode::HamtShard(_, ref hamt) => Ok(InnerNode::Node {
-                    node: hamt.root.clone(),
-                    value,
-                }),
-                UnixfsNode::RawNode(_)
-                | UnixfsNode::File(_)
-                | UnixfsNode::Directory(_)
-                | UnixfsNode::Raw(_)
-                | UnixfsNode::Symlink(_) => Ok(InnerNode::Leaf {
-                    link: link.clone(),
-                    value,
-                }),
-            },
-            OutContent::Raw(_, bytes) => {
-                let node = UnixfsNode::decode(&link.cid, bytes)?;
-
-                Ok(InnerNode::Leaf {
-                    link: link.clone(),
-                    value: node,
-                })
-            }
-            _ => bail!("unexpected node: {:?}", out.content.typ()),
+        let cid = link.cid;
+        let loaded_cid = loader.load_cid(&cid, &ctx).await?;
+        let node = UnixfsNode::decode(&cid, loaded_cid.data)?;
+        //
```

Comment:
need to handle Raw

Reply:
double checked the code path, `UnixfsNode::decode` handles "decoding" from `Raw`: 
https://github.com/n0-computer/iroh/blob/5702c3e4666557e70993ebcf90b52cf1e04f0c0b/iroh-resolver/src/unixfs.rs#L187-L189

let me know if you meant something different